viz-lib/src/visualizations/chart/Editor/ColumnMappingSelect.jsx

```diff
@@ -12,9 +12,18 @@ const MappingTypes = {
   zVal: { label: "Color Column" },
 };
 
-export default function ColumnMappingSelect({ value, availableColumns, type, onChange }) {
+export default function ColumnMappingSelect({ value, availableColumns, type, onChange, isAxesInverted }) {
   const options = sortBy(filter(uniq(flatten([availableColumns, value])), v => isString(v) && v !== ""));
-  const { label, multiple } = MappingTypes[type];
+
+  let { label, multiple } = MappingTypes[type];
+  // this inverts the ui, as the data will be inverted on render
+  if (isAxesInverted) {
+    if (type === "x") {
+      label = label.replace("X", "Y");
```

Comment:
I had some thoughts about creating a sort of "translate" method `t("x", options)` for labels. However, this is currently only for the Axis name, so it didn't seem worth it the extra logic. Leaving here as a future note 🙂.

Reply:
This is the sort of abstraction I was thinking of. Nice to hear your thoughts on it 😁

Reply:
My suggestions is to introduce another mappings e.g. `const InvertedAxesMappingTypes = extend({}, MappingTypes, ...}` and then choose appropriate mapping depending on `isAxesInverted` flag.

P.S. Since you're using plural form (Axes) in flag name - maybe, `areAxesInverted` then? Also, another thing to consider - axes are rather "swapped" than "inverted". Better namings help to read code easier 🙂